cancermuts/datasources.py

```diff
@@ -159,13 +159,13 @@ def get_sequence(self, gene_id, upid=None, upac=None, isoform=None):
                 raise ValueError(f"Failed to fetch UniProt JSON entry for {this_upac}")
 
             data = response.json()
-            alt_prods = [x for x in data.get("comments", []) if x["commentType"] == "ALTERNATIVE PRODUCTS"] 
+            alt_prods = [x for x in data["comments"] if x["commentType"] == "ALTERNATIVE PRODUCTS"] 
             is_canonical = False
             isoform_found = False
             for prod in alt_prods:
-                for iso in prod.get("isoforms", []):
-                    ids = iso.get("isoformIds", [])
-                    status = iso.get("isoformSequenceStatus", "")
+                for iso in prod["isoforms"]:
+                    ids = iso["isoformIds"]
+                    status = iso["isoformSequenceStatus"]
                     if isoform in ids:
                         isoform_found = True
                         is_canonical = (status == "Displayed")
```

Comment:
this is better, but we should handle `KeyError`s or other problems if they happen, e.g.:

```
alt_prods = [x for x in data["comments"] if x["commentType"] == "ALTERNATIVE PRODUCTS"]
if len(alt_prods) == 0: 
    raise ValueError("...:)

try:
    ids = iso[...]
except KeyError:
    raise ValueError(...)
```